source/spawn.js

```diff
@@ -30,6 +32,18 @@ export const spawnSubprocess = async (file, commandArguments, options, context)
 	}
 };
 
+// When the `shell` option is set, any command argument is concatenated as a single string by Node.js:
+// https://github.com/nodejs/node/blob/e38ce27f3ca0a65f68a31cedd984cddb927d4002/lib/child_process.js#L614-L624
+// However, since Node 24, it also prints a deprecation warning.
+// To avoid this warning, we perform that same operation before calling `node:child_process`.
+// Shells only understand strings, which is why Node.js performs that concatenation.
+// However, we rely on users splitting command arguments as an array.
+// For example, this allows us to easily detect whether the binary file is `node` or `node.exe`.
+// So we do want users to pass array of arguments even with `shell: true`, but we also want to avoid any warning.
+const concatenateShell = (file, commandArguments, options) => options.shell && commandArguments.length > 0
+	? [[file, ...commandArguments].join(' '), [], options]
+	: [file, commandArguments, options];
```

Comment:
From the Node.js source code, this should be doing exactly what Node.js is doing. So this is not breaking, and should have no changes except for not showing the deprecation warning anymore.